tools/minify/common/buildconfig.go

```diff
@@ -0,0 +1,156 @@
+package common
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/toml"
+	"github.com/influxdata/toml/ast"
+)
+
+type pluginState map[string]bool
+type BuildConfig struct {
+	states         map[string]pluginState
+	defaultInclude bool
+}
+
+func NewConfiguration(defaults bool) *BuildConfig {
+	return &BuildConfig{
+		states:         make(map[string]pluginState),
+		defaultInclude: defaults,
+	}
+}
+
+func (b *BuildConfig) ImportConfigurations(files, dirs []string) (int, error) {
+	// Gather all configuration files
+	var filenames []string
+	filenames = append(filenames, files...)
+
+	for _, dir := range dirs {
+		// Walk the directory and get the packages
+		elements, err := os.ReadDir(dir)
+		if err != nil {
+			return 0, fmt.Errorf("reading directory %q failed: %w", dir, err)
+		}
+
+		for _, element := range elements {
+			if element.IsDir() || filepath.Ext(element.Name()) != ".conf" {
```

Comment:
Does this mean we omit a configuration file name `.toml` or `.cfg`?

Reply:
Yes. This is exactly the same in Telegraf...

Reply:
Well TIL. You can tell I don't use config directories very often. No need to change this here.